config/initializers/indigo_service.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+# This initializer loads the optional configuration for:
+#   the indigo service
+
+# Specific
+validations = lambda do |config, service|
+  url = URI.parse(config.send(service)&.indigo_service_url)
+  raise ArgumentError, "Invalid URL: #{url}" unless url.host && %w[http https].include?(url.scheme)
+
+  # set description
+  config.send(service).desc = "service hosted at: #{url}"
+end
+
+# Generic initialization
+service = File.basename(__FILE__, '.rb').to_sym # Service name
+service_setter = :"#{service}=" # Service setter
+ref = "Initializing #{service}:" # Message prefix
```

Comment:
If something goes wrong with the current code error would be something like:

"Initializing #{service}:  Error while loading configuration #{e.message}"

- Service is defined: service = File.basename(__FILE__, '.rb').to_sym # Service name
- e.message: will state the complete message for what went wrong

Eventually, it would be something like this:
**"Initializing Indigo_service:  Error while loading configuration #{e.message}"**

I think there would be enough content for the developer to understand if anything goes bad!